fe/fe-core/src/main/java/com/starrocks/qe/scheduler/QueryRuntimeProfile.java

```diff
@@ -337,6 +344,15 @@ public void updateProfile(FragmentInstanceExecState execState, TReportExecStatus
             saveRunningProfile(profilingPlan, profile);
             LOG.debug("update profile, profilingPlan: {}, profile: {}", profilingPlan, profile);
         }
+
+        // Merge it into fragment-profile if this instance already finished
+        if (params.isSetProfile() && params.isDone()) {
```

Comment:
so without consider runtime profile uploading, before this pr, every fragment instance profile will keep until merged by profile-worker thread. after this pr, fragment instance profile's counters will merged into fragment immediately in here. If some instance profile upload very slow or profile-worker is super busy, then your pr will help to reduce memory?

Reply:
change the strategy to handle this case: once a fragment-instance finish, purse that profile and merge it into fragment-profile, it can reduce memory